### DISCUSSION TITLE
Visualizing remote audio streams in a WebRTC application

### DISCUSSION THREAD

**Discussion by MouDev99** [hvianna/audioMotion-analyzer#84] [Q&A] [answered]
I've built a video/audio chatting web app using PeerJS for WebRTC connections. I'm currently using audio-motion to visualize local audio from the user's microphone, which works well. However, I'm having trouble visualizing the audio from remote streams received from other peers.

Is it possible to use audio-motion to visualize these remote audio streams? If so, what's the best approach to connect a remote MediaStream to audio-motion for visualization?

My current setup:

- Local audio visualization works fine
- Remote audio plays correctly, but I can't get it to visualize
- I have access to the remote MediaStream object

Any guidance on how to properly set up audio-motion with remote WebRTC streams would be greatly appreciated. If audio-motion isn't suitable for this use case, are there any alternative libraries you'd recommend?

**Comment by hvianna** [comment, marked as answer]
It should be possible to use the remote stream as the `srcObject` of an [HTMLAudioElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAudioElement/Audio) object and connect this object to audioMotion.

```js
const audio = new Audio();
audio.srcObject = yourRemoteStream;
audioMotion.connectInput( audio );
audio.play();
```

**Comment by AllonisDave** [comment]
Has anyone got this to work with a webrtc stream? We are using the webrtc-player from Softvelum.  Analyzer gets created. Sources look to be connected. Just no visualizations. AudioContext claims it is running. Thoughts?

![a1](https://github.com/user-attachments/assets/0f649c87-62d5-4ff0-a2e1-c025fc6b5aeb)


![a2](https://github.com/user-attachments/assets/6bdb75e0-8647-4dea-a0e3-9eccd6bdf46e)
